algo/src/main/java/com/hartwig/serve/sources/ckb/CkbExtractor.java

```diff
@@ -175,23 +186,37 @@ private static ExtractionResult toExtractionResult(@NotNull String variant, @Not
             }
         }
 
-        return ImmutableExtractionResult.builder()
-                .refGenomeVersion(Knowledgebase.CKB.refGenomeVersion())
-                .addEventInterpretations(interpretation)
-                .knownHotspots(convertToKnownHotspots(output.hotspots(), variant, transcript))
-                .knownCodons(convertToKnownCodons(codons))
-                .knownExons(convertToKnownExons(output.exons()))
-                .knownGenes(output.fusionPair() == null ? convertToKnownGenes(gene) : Collections.emptySet())
-                .knownCopyNumbers(convertToKnownAmpsDels(output.copyNumber()))
-                .knownFusions(convertToKnownFusions(output.fusionPair()))
-                .actionableHotspots(actionableHotspots)
-                .actionableCodons(actionableCodons)
-                .actionableExons(actionableExons)
-                .actionableGenes(actionableGenes)
-                .actionableFusions(actionableFusions)
-                .actionableCharacteristics(actionableCharacteristics)
-                .actionableHLA(actionableHLA)
-                .build();
+        if (generateKnownEvents) {
+            return ImmutableExtractionResult.builder()
+                    .knownHotspots(convertToKnownHotspots(output.hotspots(), variant, transcript))
+                    .knownCodons(convertToKnownCodons(codons))
+                    .knownExons(convertToKnownExons(output.exons()))
+                    .knownGenes(output.fusionPair() == null ? convertToKnownGenes(gene) : Collections.emptySet())
+                    .knownCopyNumbers(convertToKnownAmpsDels(output.copyNumber()))
+                    .knownFusions(convertToKnownFusions(output.fusionPair()))
+                    .refGenomeVersion(Knowledgebase.CKB_EVIDENCE.refGenomeVersion())
+                    .addEventInterpretations(interpretation)
+                    .actionableHotspots(actionableHotspots)
+                    .actionableCodons(actionableCodons)
+                    .actionableExons(actionableExons)
+                    .actionableGenes(actionableGenes)
+                    .actionableFusions(actionableFusions)
+                    .actionableCharacteristics(actionableCharacteristics)
+                    .actionableHLA(actionableHLA)
+                    .build();
+        } else {
+            return ImmutableExtractionResult.builder()
+                    .refGenomeVersion(Knowledgebase.CKB_EVIDENCE.refGenomeVersion())
+                    .addEventInterpretations(interpretation)
+                    .actionableHotspots(actionableHotspots)
+                    .actionableCodons(actionableCodons)
+                    .actionableExons(actionableExons)
+                    .actionableGenes(actionableGenes)
+                    .actionableFusions(actionableFusions)
+                    .actionableCharacteristics(actionableCharacteristics)
+                    .actionableHLA(actionableHLA)
+                    .build();
+        }
```

Comment:
There is duplication here!

Reply:
Learned from you how I could solve this :)